src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
+        if ((this->actor.params & 3) == 2) {
+            // This function is blank
+            func_80A9AFA0(this, func_80A9BF30);
+        }
+    } else {
+        func_80A9AFA0(this, func_80A9BEFC);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BBB0.s")
+void func_80A9BEFC(EnKusa* this, GlobalContext* globalCtx) {
+    if (this->unk_19C >= 0x78) {
+        func_80A9C00C(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BC1C.s")
+void func_80A9BF30(EnKusa* this, EnKusaActionFunc actionFunc) {
+    // Function was intentionally left blank?...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BEAC.s")
+void func_80A9BF3C(EnKusa* this) {
+    this->actor.posRot.pos.x = this->actor.initPosRot.pos.x;
+    this->actor.posRot.pos.y = this->actor.initPosRot.pos.y - 9.0f;
+    this->actor.posRot.pos.z = this->actor.initPosRot.pos.z;
+    // Set scale factors
+    func_80A9B1FC(this);
+    this->actor.shape.rot = this->actor.initPosRot.rot;
+    func_80A9AFA0(this, func_80A9BFA8);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BEFC.s")
+void func_80A9BFA8(EnKusa* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BF30.s")
+    if (this->unk_19C >= 0x79) {
+        if ((Math_ApproxF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y, 0.6f) != 0) &&
+            (this->unk_19C >= 0xAA)) {
+            func_80A9C00C(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BF3C.s")
+void func_80A9C00C(EnKusa* this) {
+    func_80A9AFA0(this, func_80A9C068);
+    func_80A9B1FC(this);
+    this->actor.shape.rot = this->actor.initPosRot.rot;
+    this->actor.flags = this->actor.flags & -0x801;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BFA8.s")
+void func_80A9C068(EnKusa* this, GlobalContext* globalCtx) {
+    s32 sp24;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9C00C.s")
+    sp24 = Math_ApproxF(&this->actor.scale.y, 0.4f, 0.014f) & 1;
+    sp24 &= Math_ApproxF(&this->actor.scale, 0.4f, 0.011f);
+    this->actor.scale.z = this->actor.scale.x;
+    if (sp24 != 0) {
+        Actor_SetScale(this, 0.4f);
+        func_80A9B89C(this);
+        this->collider.base.maskA = this->collider.base.maskA & 0xFFF7;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9C068.s")
+void EnKusa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/EnKusa_Update.s")
+    this->unk_19C = this->unk_19C + 1;
+    this->actionFunc(this, globalCtx);
+    if ((this->actor.flags & 0x800) != 0) {
+        this->actor.shape.unk_08 = -6.25f;
+        return;
+    }
+    this->actor.shape.unk_08 = 0.0f;
```

Comment:
```suggestion
    if ((this->actor.flags & 0x800) != 0) {
        this->actor.shape.unk_08 = -6.25f;
    } else {
        this->actor.shape.unk_08 = 0.0f;
    }
```
Could also be a ternary: `this->actor.shape.unk_08 = (this->actor.flags & 0x800) ? -6.25f: 0.0f;